### ISSUE TITLE
[Bug] Opacity problem on stl file when backface culling is enabled

### ISSUE THREAD

**Issue by TimDeBauwReLu** [Kitware/vtk-js#3069]
### Bug description

I have a problem where the opacity of stl objects behaves inconsistently when backface culling is enabled. I have two meshes of a mandible, which both are rendered correctly when opacity=1 (see pictures below). I also checked that the backfaces of both meshes are on the inside.
<img width="705" alt="Screenshot 2024-05-13 at 12 58 00" src="https://github.com/Kitware/vtk-js/assets/114484998/c696289c-6895-477a-95aa-4d10cc8f618c">
<img width="788" alt="Screenshot 2024-05-13 at 12 58 11" src="https://github.com/Kitware/vtk-js/assets/114484998/4986b1ec-c04c-4518-89a3-8cd8a8ec9a88">

When changing the opacity to be less than 1, you can see that the color of the bone is different on both objects (see pictures below). When zooming out far, both objects are again rendered correctly. Do you know what can cause this?

<img width="863" alt="Screenshot 2024-05-13 at 12 57 22" src="https://github.com/Kitware/vtk-js/assets/114484998/a348fead-ebf8-4c42-b2d8-269ecd03ccf6">
<img width="818" alt="Screenshot 2024-05-13 at 12 57 41" src="https://github.com/Kitware/vtk-js/assets/114484998/04cd95ac-057a-474a-947f-eecadedec98a">

### Steps to reproduce

```
const reader = vtkSTLReader.newInstance()
reader.setUrl('Mandible_broken.stl', { binary: true })

const mapper = vtkMapper.newInstance();
mapper.setInputConnection(reader.getOutputPort());

const actor = vtkActor.newInstance();
actor.getProperty().setBackfaceCulling(true);
actor.getProperty().setOpacity(0.5);
actor.setMapper(mapper);

const renderer = fullScreenRenderer.getRenderer();
const renderWindow = fullScreenRenderer.getRenderWindow();
renderer.addActor(actor);
 renderWindow.render();
```
        

### Expected Behavior

Both objects are rendered correctly when opacity is less than 1 and backface culling is enabled

### Environment

- vtk.js version:
- Browsers:
- OS:

**Comment by finetjul** [comment]
What picture is incorrect ? do you have backface culling enabled on the incorrect picture ?

**Comment by TimDeBauwReLu** [comment]
I forgot to mention that this problem does not appear when using `camera.setParallelProjection(true);`

**Comment by finetjul** [comment]
What's the difference between the 3rd picture and the last picture ? Only the mesh is different ?

**Comment by TimDeBauwReLu** [comment]
The color of the bone is too dark, and you can see a big contrast between the bone and the teeth sockets (dark red vs bright red).  It is supposed to be like this I believe: 
<img width="342" alt="Screenshot 2024-05-13 at 14 15 38" src="https://github.com/Kitware/vtk-js/assets/114484998/8f0f461a-08ad-4d15-9420-b939b5411104">

**Comment by finetjul** [comment]
Gotcha.

It would be great if you could attach that STL file to the issue. 
The shader code should be investigated...

**Comment by TimDeBauwReLu** [comment]
Here are the files used to reproduce it:

[Archive.zip](https://github.com/Kitware/vtk-js/files/15295153/Archive.zip)

**Comment by finetjul** [comment]
You might want to check the magnitude of your normals (maybe it is different between the teeth triangles and the jaw triangles). 
Alternatively, you might also want to test without diffuse light coefficient, only ambient, and see if the problem is light related.
Finally, you might want to consider the OrderIndependentTranslucentPass code...

**Comment by TimDeBauwReLu** [comment]
@finetjul It looks like it has something to do with the diffuse light coefficient. When I put it to zero, it looks like this.

<img width="831" alt="Screenshot 2024-05-15 at 17 04 18" src="https://github.com/Kitware/vtk-js/assets/114484998/cbe9bed5-0287-4479-981e-d28ae946b5b6">

I guess that depending on the distance from the camera to the object, the intensity of the diffuse component changes? Do you have an idea where to investigate this issue?

**Comment by finetjul** [comment]
And this blackness does not happen if it is opaque? Do you have ambient lighting?

**Comment by TimDeBauwReLu** [comment]
Yes, it is caused because I do not have ambient lighting I believe
